src/content/docs/cypher/syntax.md

```diff
@@ -0,0 +1,196 @@
+
+```cypher
+MATCH (a:Person)
+WHERE a.age < 30
+RETURN a.*;
+```
+
+## Clauses
+
+A Cypher query may contain one or more clauses and their associated subclauses, and can span multiple lines.
+The end of a statement is marked with a semicolon `;`, and the query parser looks for this symbol
+to know when a statement is complete.
+
+Examples of clauses include:
+- `MATCH`: Find patterns in the graph
+- `RETURN`: Specify what subset of the matched data to return
+
+Examples of subclauses (that must reside under a clause) include:
+- `WHERE`: Filter the results of a `MATCH` clause
+- `LIMIT`: Limit the number of results returned by a query
+
+## Comments
+
+Comments are for humans to read and document their code, and are ignored by the
+query parser.
+- Single line comments begin with a double slash (`//`) and continue up until the end of the
+line. They can be placed at the beginning, in the middle, or at the end of a query.
+- Multi-line comments begins with a slash and asterisk (`/*`) and continues until it ends with an
+asterisk and a slash (`*/`). They can be useful for comments that are too long for one line.
+
+Some examples are below.
+
+```cypher
+// Whole-line comment before a query
+MATCH (a:Person) RETURN a.*
+```
+
+```cypher
+MATCH (a:Person) RETURN a.*  // Comment at the end of a query
+```
+
+```cypher
+MATCH (a:Person)
+// Comment in the middle of a query
+WHERE a.age < 30
+RETURN a.*
+```
+
+```sql
+/*
+This is a comment
+spanning multiple lines
+*/
+MATCH (a:Person) RETURN a.* 
+```
+
+## Naming rules and recommendations
+
+As a general rule of thumb, ensure the following:
+- Names should begin with an valid alphabetic character of type unicode string -- `Person`, `CarOwner` 
+- Names should not begin with a number -- `1Person` is invalid, but `Person1` is valid
+- Names should not contain whitespaces or special characters other than underscores -- `CarOwner` is valid, but `Car Owner` is invalid
+- Names are generally case-insensitive -- `Person` is the same as `person`, during table creation and querying
+
+The following naming conventions are recommended for node and relationship tables:
+
+| Type | Naming convention | Do | Don't |
+|---|---|---|---|
+| Node tables | CamelCase (begin with upper case letter) | `CarOwner` | `car_owner` |
+| Relationship tables | CamelCase or UPPERCASE separated by underscores | `IsPartOf`/`IS_PART_OF` | `isPartOf` or `is_part_of` |
+
+## Parameters
+
+Parameters in Cypher queries are placeholders for values that are provided at runtime.
+Parameters are prefixed with a dollar sign `$` and can be used in any part of a query. They are useful for
+preventing Cypher injection attacks, and for reusing query templates with different values.
+
+See the [prepared statements](/get-started/prepared-statements) guide for more information on how to use parameters in Kùzu.
+
+## Reserved keywords
+
+Reserved keywords are words that have a special meaning in Cypher. They cannot be used as identifiers
```

Comment:
First I think we should call this section keyword. And inside we have reserved and non-reserved keyword.

We will highlight all keywords. Non-reserved ones can still used as variables.

keyword list is in `src/antlr/keywords.txt`
non-reserved ones is in `src/antlr/Cypher.g4` and rule `kU_NonReservedKeywords`

I would just follow [PG](https://www.postgresql.org/docs/current/sql-keywords-appendix.html) and organize them in alphabet order

Reply:
Is it just as important to list non-reserved keywords? if it's not reserved, why should the user be told that it can be used as variables? This is redundant and expects the user to do more work, imo. Also, I agree with Semih in that PG shouldn't be brought in here unless absolutely necessary (it creates more confusion because this is for a Cypher manual).

Reply:
Again, this section is designed to be more or less identical to the openCypher9 PDF:
https://s3.amazonaws.com/artifacts.opencypher.org/openCypher9.pdf

Reply:
> We will highlight all keywords.
This point, I agree, makes sense. I'll think a bit more about how/where to list all available keywords. But I think the reserved keywords subheader should be present as this is much more important for the reader.